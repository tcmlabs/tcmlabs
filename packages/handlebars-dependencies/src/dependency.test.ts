import * as HandlebarsAnalyser from '.';

describe('Dependency graph', () => {
  test('Retrieve a single template dependencies', () => {
    const template = `
      {{ foo }}
      {{ baz.duh }}
      {{ missing.quxProperty }}
    `;

    const dependencies = HandlebarsAnalyser.templateDependencies(template);

    expect(dependencies).toMatchSnapshot();
  });

  test('Retrieve multiple template names', () => {
    const templates = [{ name: 'foo', content: '{{ bar }}' }];

    const components = HandlebarsAnalyser.templateNames(templates);

    expect(components).toMatchSnapshot();
  });

  test('Retrieve template existing dependencies', () => {
    const template = `
      {{ foo }}
      {{ baz.duh }}
      {{ missing.quxProperty }}
    `;

    const templates = [{ name: 'foo', content: '{{ bar }}' }];

    const existingDependencies = HandlebarsAnalyser.existingDependencies(templates, template);

    expect(existingDependencies).toMatchSnapshot();
  });

  test('Retrieve template missing dependencies', () => {
    const template = `
      {{ foo }}
      {{ baz.duh }}
      {{ missing.quxProperty }}
    `;

    const templates = [
      { name: 'foo', content: '{{ bar }}' },
      { name: 'baz', content: '{{ qux }}' },
    ];

    const missingDependencies = HandlebarsAnalyser.missingDependencies(templates, template);

    expect(missingDependencies).toMatchSnapshot();
  });

  test('Analyse dependencies', () => {
    const template = `
      {{ foo }}
      {{ baz.duh }}
      {{ missing.quxProperty }}
    `;

    const templates = [
      { name: 'foo', content: '{{ bar }}' },
      { name: 'baz', content: '{{ qux }}' },
    ];

    const analysedDependencies = HandlebarsAnalyser.markDependencyPresences(templates, template);

    expect(analysedDependencies).toMatchSnapshot();
  });
});
