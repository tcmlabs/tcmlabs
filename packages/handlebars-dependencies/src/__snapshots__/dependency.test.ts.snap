// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependency graph Analyse dependencies 1`] = `
Array [
  Object {
    "exist": true,
    "name": "foo",
  },
  Object {
    "exist": true,
    "name": "baz",
  },
  Object {
    "exist": false,
    "name": "missing",
  },
]
`;

exports[`Dependency graph Retrieve a single template dependencies 1`] = `
Array [
  "foo",
  "baz",
  "missing",
]
`;

exports[`Dependency graph Retrieve multiple template names 1`] = `
Array [
  "foo",
]
`;

exports[`Dependency graph Retrieve template existing dependencies 1`] = `
Array [
  "foo",
]
`;

exports[`Dependency graph Retrieve template missing dependencies 1`] = `
Array [
  "missing",
]
`;
